components/brave_wallet/common/zcash_utils.cc

```diff
@@ -19,24 +19,25 @@
 #include "brave/components/brave_wallet/common/hash_utils.h"
 #include "brave/components/brave_wallet/rust/lib.rs.h"
 #include "brave/third_party/bitcoin-core/src/src/base58.h"
+#include "brave/third_party/bitcoin-core/src/src/bech32.h"
+#include "brave/third_party/bitcoin-core/src/src/util/strencodings.h"
 
 namespace brave_wallet {
 
 namespace {
+constexpr char kTestnetHRP[] = "utest";
+constexpr char kMainnetHRP[] = "u";
 constexpr size_t kPaddedHrpSize = 16;
 constexpr size_t kPubKeyHashSize = 20;
 constexpr size_t kPrefixSize = 2;
 
-// https://zips.z.cash/zip-0316#encoding-of-unified-addresses
-enum AddrType {
-  kP2PKH = 0x00,
-  kP2PSH = 0x01,
-  kSapling = 0x02,
-  kOrchard = 0x03,
-  kMaxValue = kOrchard
-};
-
-using ParsedAddress = std::pair<AddrType, std::vector<uint8_t>>;
+std::array<uint8_t, kPaddedHrpSize> GetPaddedHRP(bool is_testnet) {
+  std::string hrp = is_testnet ? kTestnetHRP : kMainnetHRP;
+  std::array<uint8_t, kPaddedHrpSize> padded_hrp;
+  std::fill(std::begin(padded_hrp), std::end(padded_hrp), 0);
+  base::ranges::copy(base::make_span(hrp), padded_hrp.begin());
```

Comment:
we could static_assert that kPaddedHrpSize is larger than sizeof(kTestnetHRP) etc